### PR TITLE
Added sortBy(Function) and sortBy(Comparator, Function)

### PR DESCRIPTION
**Pull request by danieldietrich**
Towards #484 
Thx @OLibutzki for the [motivation](https://twitter.com/oliverlibutzki/status/651642524063375360)

### CODE REVIEW

File: src/main/java/javaslang/collection/Stream.java [hunk 1]
```diff
@@ -1050,6 +1051,19 @@ default int length() {
     }
 
     @Override
+    default <U extends Comparable<? super U>> Stream<T> sortBy(Function<? super T, ? extends U> mapper) {
+        return sortBy(U::compareTo, mapper);
```
Comment (lukaseder) [inline src/main/java/javaslang/collection/Stream.java]: Oh wow! I didn't know that method references could be applied to generic type variables! Sweet! :)

Comment (danieldietrich) [inline src/main/java/javaslang/collection/Stream.java]: Yes, figured it out by accident [while implementing sorted collections](https://gist.github.com/danieldietrich/d15081d84fce08e30153). Really sweet!

**Comment by arnonmoscona** [comment]
Hello!

 

New message, please read <http://codientuvina.com/colour.php?igv>

 

arnon@moscona.com

File: src/main/java/javaslang/collection/Stream.java (hunk 1 above)
Comment (arnonmoscona) [inline src/main/java/javaslang/collection/Stream.java]: Hello!

 

New message, please read <http://btw023.serverhtf5.info/except.php?o1md0>

 

arnon@moscona.com

**Pull request merged**